File: NowPlaying.m
//
// NowPlaying.m
// Now Playing Cordova Plugin Version 2
// Created by Koiwu Beyan on 11/14/2013.
//

#import "NowPlaying.h"
#import <MediaPlayer/MediaPlayer.h>
#import <MediaPlayer/MPNowPlayingInfoCenter.h>
#import <MediaPlayer/MPMediaItem.h>
#import <AVFoundation/AVFoundation.h>

@implementation NowPlaying


- (void)updateMetas:(CDVInvokedUrlCommand*)command

{
	
	NSString *artist = [command.arguments objectAtIndex:1];
	NSString *title = [command.arguments objectAtIndex:2];
	NSString *album = [command.arguments objectAtIndex:3];
    
    MPNowPlayingInfoCenter *infoCenter = [MPNowPlayingInfoCenter defaultCenter];
    infoCenter.nowPlayingInfo = [NSDictionary dictionaryWithObjectsAndKeys:
                                 artist, MPMediaItemPropertyArtist,
                                 title, MPMediaItemPropertyTitle,
                                 album, MPMediaItemPropertyAlbumTitle,
                                 [[MPMediaItemArtwork alloc] initWithImage:[UIImage imageNamed:@"albumart.png"]], MPMediaItemPropertyArtwork,
                                 nil];
    
}

@end

File: NowPlaying.js
//
//  NowPlaying.js
//	Now Playing Cordova Plugin
//	Update by Koiwu Beyan
//	Created by François LASSERRE on 12/05/13.
//	Copyright 2013 François LASSERRE. All rights reserved.
//  MIT Licensed
//
(function(){

	var cordovaRef = window.PhoneGap || window.Cordova || window.cordova; // old to new fallbacks
	
	var NowPlaying = function() {};
	
	/**
	 * Update metadatas.
	 */
	NowPlaying.prototype.updateMetas = function(artist,title,album) {
		cordovaRef.exec(null, null, "NowPlaying", "updateMetas", [artist,title,album]);
	};
	
	cordovaRef.addConstructor(function(){
		if (!window.plugins) {
			window.plugins = {};
		}
		window.plugins.nowPlaying = new NowPlaying();
	});
	
})();
